api/cloud-control/v1beta1/nuke_types.go

```diff
@@ -61,22 +61,32 @@ func (s *NukeStatus) GetKindNoCreate(kind string) *NukeStatusKind {
 	return nil
 }
 
-func (s *NukeStatus) GetKind(kind string) (*NukeStatusKind, bool) {
+func (s *NukeStatus) GetKind(kind string, resourceType ResourceType) (*NukeStatusKind, bool) {
 	result := s.GetKindNoCreate(kind)
 	if result != nil {
 		return result, false
 	}
 	result = &NukeStatusKind{
-		Kind:    kind,
-		Objects: map[string]NukeResourceStatus{},
+		Kind:         kind,
+		ResourceType: resourceType,
+		Objects:      map[string]NukeResourceStatus{},
 	}
 	s.Resources = append(s.Resources, result)
 	return result, true
 }
 
+type ResourceType string
+
+const (
+	KcpManagedResource ResourceType = "KCP"
+	ProviderResource   ResourceType = "Provider"
+)
+
 type NukeStatusKind struct {
 	Kind    string                        `json:"kind"`
 	Objects map[string]NukeResourceStatus `json:"objects"`
+	// +optional
```

Comment:
Default it to KCP so once new CRD is applied, already existing resources hopefully get the "KCP" value in that field.
```suggestion
	// +optional
	// +kubebuilder:default=KCP
```

Reply:
Done